src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1484,12 +1484,32 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     return;
   }
 
-  AppIdURL app_url = policy_manager_->GetNextUpdateUrl(urls);
-  while (!IsUrlAppIdValid(app_url.first, urls)) {
-    app_url = policy_manager_->GetNextUpdateUrl(urls);
+  const uint32_t app_id = GetAppIdForSending();
+  const ApplicationSharedPtr app = application_manager_.application(app_id);
+  const std::string app_policy_id =
+      app != nullptr ? app->policy_app_id() : policy::kDefaultId;
```

Comment:
@ZhdanovP Use http://www.cplusplus.com/reference/memory/shared_ptr/operator%20bool/ instead of comparison